src/utils/config_filters.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+	Copyright (C) 2003 - 2023
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "config.hpp"
+
+/**
+ * Utility functions for implementing [filter], [filter_ability], [filter_weapon], etc.
+ *
+ * For example, a filter of `x=1` puts a requirement on the value of `x` but accepts any value of `y`.
+ */
+namespace utils::config_filters
+{
+/**
+ * Checks whether the filter matches the value of @a cfg[@a attribute]. If @a cfg doesn't have that
+ * attribute, assume that an unset value is equivalent to @a def.
+ *
+ * Always returns true if the filter puts no restriction on the value of @a cfg[@a attribute].
+ */
+bool bool_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool def);
+bool double_matches_if_present(const config& filter, const config& cfg, const std::string& attribute);
+
+bool int_matches_if_present_positive(const config& filter, const config& cfg, const std::string& attribute);
+bool int_matches_if_present_signed(const config& filter, const config& cfg, const std::string& attribute);
```

Comment:
The function names are getting a bit long, I'm open to suggestions.

Also, just noticed that this one still needs documentation.

Reply:
I've gone for a rename to this trio:

* int_matches_if_present
* int_matches_if_present_or_negative
* unsigned_matches_if_present